mllib/src/main/scala/org/apache/spark/ml/tree/impl/RandomForest.scala

```diff
@@ -1001,11 +996,18 @@ private[spark] object RandomForest extends Logging {
     } else {
       val numSplits = metadata.numSplits(featureIndex)
 
-      // get count for each distinct value
-      val (valueCountMap, numSamples) = featureSamples.foldLeft((Map.empty[Double, Int], 0)) {
+      // get count for each distinct value except zero value
+      val (partValueCountMap, partNumSamples) = featureSamples.foldLeft((Map.empty[Double, Int], 0)) {
```

Comment:
This bit of (existing) code seems obtuse to me. What about...

```
val partNumSamples = featureSamples.size
val partValueCountMap = scala.collection.mutable.Map[Double,Int]()
featureSamples.foreach { x =>
  partValueCountMap(x) = partValueCountMap.getOrElse(x, 0) + 1
}
```

Reply:
have fixed it.